runtime/lua/vim/lsp/structures/location.lua

```diff
@@ -0,0 +1,154 @@
+local highlight = require('vim.highlight')
+local util = require('vim.lsp.util')
+
+local Position = require('vim.lsp.structures.position')
+
+local api = vim.api
+
+local get_first_or_only = function(item)
+  if item == nil or type(item) ~= 'table' then
+    return item
+  end
+
+  if not vim.tbl_islist(item) then
+    return item
+  else
+    return item[1]
+  end
+end
+
+--- Set of functions for working with `Location` objects.
+--- This set of functions also handles working with `LocationLink` for convenience.
+---
+--@ref https://microsoft.github.io/language-server-protocol/specifications/specification-current/#location
+--@ref https://microsoft.github.io/language-server-protocol/specifications/specification-current/#locationLink
+local Location = {}
+
+Location.to_uri_and_range = function(location)
+  local uri = location.targetUri or location.uri
```

Comment:
Should probably assert that if it has targetUri then it also has other target? Not sure if we need to do that much validation on messages though...